backend/services/implementations/classService.ts

```diff
@@ -191,6 +193,32 @@ class ClassService implements IClassService {
     }
   }
 
+  async archiveClass(id: string): Promise<ClassResponseDTO> {
+    let archivedClass: Class | null;
+    try {
+      archivedClass = await MgClass.findOneAndUpdate(
+        { _id: id },
```

Comment:
do we need to add this? i think we should probably throw the appropriate error if the class is already archived...
```suggestion
        { _id: id, isActive: true },
```